assets/js/modal.js

```diff
@@ -31,12 +32,36 @@ function renderModal () {
   })
 }
 
+/**
+ * Trap focus in modal
+ * Only called on open modals
+ */
+function trapFocus (e) {
```

Comment:
I think a more reliable way would be to let the focus work normally, but have a `focus` listener that moves the focus back to the modal once it falls outside of it. [This page](https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html) looks like a reference example and the interesting bit is [here](https://github.com/w3c/aria-practices/blob/3dd5a2d3af89adc1249bc76106eefce0e44a5de5/examples/dialog-modal/js/dialog.js#L288-L310) :)

Reply:
I tried this, but even though it looks more bullet prove it seems to be broken on safari (15). For some reason `event.target` on the `focus` event seems to always point to `.search-input`.

Reply:
Interesting, does the example from the original page work for you in safari or same problem?

Reply:
The previous implementation does indeed work, but I can surely see the issue in trying to reimplement focus handling.

Reply:
I meant the ["Add Delivery Address
" example](https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html#ex_label) :D

Reply:
I can't reproduce it on the "Add deliver address" modal, but on the second level modal when clicking on "Verify address, where focus jumps between the first and the last targetable element. Possibly because the main page does not include form elements on its own.

Reply:
So in our case if you manually focus an element (let's say the checkbox), then hitting Tab moves the focus to the search form?

Reply:
It at least tries to (`event.target` points to the search), then the event handler comes in and either moves the focus to the first or last focusable element in the modal (the close button or the live book url input).

Reply:
Could it be [this](https://stackoverflow.com/a/1914496)?

Reply:
😩 that's it. In the German translation it's even worse because it sounds more about being about highlighting things, rather than targeting. Sorry for the noice, but I'm not using safari again for more than a few days.

Reply:
No worries, thanks for confirming! Now we know what causes this behaviour and I think it's fine to stick to the current approach :)